src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,34 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public int hashCode() {
+        int result = 17;
+        result = 31 * result + ((width == null) ? 0 : width);
+        result = 31 * result + ((length == null) ? 0 : length);
+        result = 31 * result + ((color == null) ? 0 : color.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+
+        if (object == null || !object.getClass().equals(Rectangle.class)) {
+            return false;
+        }
+
+        Rectangle rectangle = (Rectangle) object;
+        return (width == null || length == null || color == null)
+                ? rectangle.width == null
+                    || rectangle.length == null
+                    || rectangle.color == null
+                        : (width == rectangle.width)
+                        || (length == rectangle.length)
+                        || (color.equals(rectangle.color));
```

Comment:
thats complicated and hard to read. Let's  check each pair separately one by one